qiskit/primitives/__init__.py

```diff
@@ -17,59 +17,181 @@
 
-.. currentmodule:: qiskit.primitives
+Migration from Primitives V1 to V2
+==================================
 
-Estimator
-=========
+The formal distinction between the Primitives V1 and V2 APIs are the base classes from which 
+primitives implementations inherit, which are all listed at the bottom of the page. At a conceptual 
+level, however, here are some notable differences keep in mind when migrating from V1 to V2:
+
+1. The V2 primitives favour vectorized inputs, where single circuits can be grouped with 
+   vector-valued (or more generally, array-valued) specifications. Each group is called a 
+   primitive unified bloc (pub), and each pub gets its own result. For example, in the estimator, 
+   you can compare the following differences:
+
+   .. code-block:: python
+
+      # Favoured V2 pattern. There is only one pub here, but there could be more.
+      job = estimator_v2.run([(circuit, [obs1, obs2, obs3, obs4])])
+      evs = job.result()[0].data.evs
+
+      # V1 equivalent, where the same circuit must be provided four times.
+      job = estimator_v1.run([circuit, circuit, circuit, circuit], [obs1, obs2, obs3, obs4])
+      evs = job.result().values
+
+   Not shown in the above example, for brevity, is that the circuit can be parametric, with arrays
+   of parameter value sets broadcasted against the array of observables. The sampler is similar,
+   but with no observables:
+
+   .. code-block:: python
+
+      # Favoured V2 pattern. There is only one pub here, but there could be more.
+      job = sampler_v2.run([(circuit, [vals1, vals2, vals3])])
+      samples = job.result()[0].data
+
+      # V1 equivalent, where the same circuit must be provided three times.
+      sampler_v1.run([circuit, circuit, circuit], [vals1, vals2, vals3])
+      quasi_dists = job.result().quasi_dists
+
+2. The V2 sampler returns samples of classical outcomes, preserving the shot order in which they 
+   were measured. This is in contrast to the V1 sampler that outputs quasi-probability distributies
+   which are instead an *estimate of the distribution* over classical outcomes. Moreover, the V2
+   sampler result objects organize data in terms of their input circuits' classical register 
+   names, which provides natural compatibility with dynamic circuits.
+
+   .. code-block:: python
+
+      circuit = QuantumCircuit(QuantumRegister(2, "qreg"), ClassicalRegister(2, "alpha"))
+      circuit.h(0)
+      circuit.cx(0, 1)
+      circuit.measure([0, 1], [0, 1])
+
+      # V1 sampler usage
+      result = sampler_v1.run([circuit]).result()
+      quasi_dists = bitstrings.get_counts()
+
+      # V2 sampler usage
+      result = sampler_v2.run([circuit]).result()
+      # these are all the bitstrings from the alpha register
+      bitstrings = result[0].data.alpha
+      # we can use it to generate a Counts mapping, which is the most similar thing to a quasi dist
+      counts = bitstrings.get_counts()           
```

Comment:
I thought it would be nice to show bitstrings as well as counts. So, I made #11700 to get bitstrings as str though it's not a strong opinion.

Reply:
Done in [ed11f05](https://github.com/Qiskit/qiskit/pull/11589/commits/ed11f053dfc79b020a5df9d2169a487399a6fe79)